docs/design/README.md

```diff
@@ -0,0 +1,595 @@
+These are built using the host-installed Rust toolchain instead and not the SDK.
+Once these are built, they are used elsewhere in the build process.
+Buildsys, for example, is the tool that launches container build environments from `build.rs` scripts.
+
+[How the Bottlerocket build system works]: https://aws.amazon.com/blogs/opensource/how-the-bottlerocket-build-system-works/
+[cargo make]: https://github.com/sagiegurari/cargo-make
+[bottlerocket-sdk]: https://github.com/bottlerocket-os/bottlerocket-sdk
+[buildsys]: https://github.com/bottlerocket-os/bottlerocket/blob/3af909865c4da765f4afc787aaf5f50eea5c3989/tools/buildsys/Cargo.toml#L2
+[pubsys]: https://github.com/bottlerocket-os/bottlerocket/blob/3af909865c4da765f4afc787aaf5f50eea5c3989/tools/pubsys/Cargo.toml#L2
+[testsys]: https://github.com/bottlerocket-os/bottlerocket/blob/3af909865c4da765f4afc787aaf5f50eea5c3989/tools/testsys/Cargo.toml#L2
+
+### Enhancing the Build System
+
+The existing build system will remain intact with certain features and abstractions added to it.
+Twoliter will serve as a facade providing a new interface over the existing build system.
+Twoliter will understand how to start a Bottlerocket build using the OOTB user's sources, packages etc.
+It will inject these directories into the existing build system by setting existing and new `Makefile.toml` variables.
+
+Twoliter will eliminate the need for all build-host software other than `docker`.
+A bootstrapping container will be published that contains any software needed to kick off and sustain a `cargo make` build command.
+
+## Kits
+
+A central idea in this design is that the build system will be able to produce and consume Docker images that contain RPM packages.
+We call these images *kits*.
+A kit is a Yum repo distributed in its entirety, atomically, as a container image.
+
+Maintainers may reference these kits in order to add packages to their Bottlerocket variants.
+In particular, the core project will publish all of its packages in a suite of kits.
+With these, an out-of-tree maintainer will have all the packages available that go into variants that the core project is currently producing.
+We envision maintainers other than the core team publishing Bottlerocket kits in the future.
+
+### Kit Signing
+
+In order to satisfy self-signing requirements, Kit RPMs will need to be GPG-signed.
+The kit will be distributed with the GPG public key, which means we need the container to also be signed.
+Existing container-signing mechanisms will be made available.
+(**TODO** - refine the design for signing artifacts, see https://github.com/bottlerocket-os/twoliter/issues/5)
+
+### Local and External Kits
+
+During the build process, there will be a distinction between *local kits* and *external kits*.
+A *local kit* is any kit where the packages are defined in the project tree and the maintainer wants to build them from source as part of the Bottlerocket build.
```

Comment:
If I understand correctly, I could have a private package which is responsible or building this kit? This would prevent me from having to have a dependency on an external container registry for the build of my variant?

Reply:
> If I understand correctly, I could have a private package which is responsible or building this kit? This would prevent me from having to have a dependency on an external container registry for the build of my variant?

You will need to depend on a container registry in order to depend on kits (and the packages they contain) from the Bottlerocket core project. Mirroring of these will be a necessary for Enterprises that cannot reach out to the Internet. For the moment we are assuming the mirrors can be configured with containerd or docker.